### ISSUE TITLE
[ BUG ] Element type is invalid: expected a string when setting components.json file

### ISSUE THREAD

**Issue by pedromchd** [mrzachnugent/react-native-reusables#250]
After setting up NativeWind/RNR alongside my Expo project, when I install a RNR component, after following the promps I get a components.json file, but this file causes the app's rendering to break:

```json
{
  "platforms": "universal",
  "aliases": {
    "components": "~/components",
    "lib": "~/lib"
  }
}
```

Error:

```
Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.

Check the render method of `AppLayout`.

This error is located at:
    in AppLayout
    in Unknown (created by Route((app)))
    in Suspense (created by Route((app)))
    in Route (created by Route((app)))
    in Route((app)) (created by SceneView)
    in StaticContainer
    in EnsureSingleNavigator (created by SceneView)
    in SceneView (created by SceneView)
    in RCTView (created by View)
    in View (created by CssInterop.View)
    in CssInterop.View (created by DebugContainer)
    in DebugContainer (created by MaybeNestedStack)
    in MaybeNestedStack (created by SceneView)
    in RCTView (created by View)
    in View (created by CssInterop.View)
    in CssInterop.View (created by SceneView)
    in RNSScreen (created by Animated(Anonymous))
    in Animated(Anonymous) (created by InnerScreen)
    in Suspender (created by Freeze)
    in Suspense (created by Freeze)
    in Freeze (created by DelayedFreeze)
    in DelayedFreeze (created by InnerScreen)
    in InnerScreen (created by Screen)
    in Screen (created by SceneView)
    in SceneView (created by NativeStackViewInner)
    in Suspender (created by Freeze)
    in Suspense (created by Freeze)
    in Freeze (created by DelayedFreeze)
    in DelayedFreeze (created by ScreenStack)
    in RNSScreenStack (created by ScreenStack)
    in ScreenStack (created by NativeStackViewInner)
    in NativeStackViewInner (created by NativeStackView)
    in RCTView (created by View)
    in View (created by CssInterop.View)
    in CssInterop.View (created by SafeAreaProviderCompat)
    in SafeAreaProviderCompat (created by NativeStackView)
    in NativeStackView (created by NativeStackNavigator)
    in PreventRemoveProvider (created by NavigationContent)
    in NavigationContent
    in Unknown (created by NativeStackNavigator)
    in NativeStackNavigator
    in Unknown (created by RootLayout)
    in ThemeProvider (created by RootLayout)
    in SessionProvider (created by RootLayout)
    in QueryClientProvider (created by RootLayout)
    in RootLayout
    in Unknown (created by Route())
    in Suspense (created by Route())
    in Route (created by Route())
    in Route() (created by ContextNavigator)
    in RNCSafeAreaProvider (created by SafeAreaProvider)
    in SafeAreaProvider (created by wrapper)
    in wrapper (created by ContextNavigator)
    in EnsureSingleNavigator
    in BaseNavigationContainer
    in ThemeProvider
    in NavigationContainerInner (created by ContextNavigator)
    in ContextNavigator (created by ExpoRoot)
    in ExpoRoot (created by App)
    in App (created by ErrorOverlay)
    in ErrorToastContainer (created by ErrorOverlay)
    in ErrorOverlay (created by withDevTools(ErrorOverlay))
    in withDevTools(ErrorOverlay)
    in RCTView (created by View)
    in View (created by CssInterop.View)
    in CssInterop.View (created by AppContainer)
    in RCTView (created by View)
    in View (created by CssInterop.View)
    in CssInterop.View (created by AppContainer)
    in AppContainer
    in main(RootComponent), js engine: hermes
```

I have no such AppLayout, since my app's entrypoint is a RootLayout function in `app/_layout.tsx`, the only way to overcome this error is removing the file, though I have to always redo de prompt in every component added and always remember to delete it before `npx expo start`.

**Comment by mrzachnugent** [comment]
Hey @pedromchd , is your layout component in `app/_layout.tsx` exported as a Default?

If so, can you please provide a minimal reproduction repo?

**Comment by pedromchd** [comment]
Sure! Here they are:

Me installing a reusable through CLI and settings components.json
![Recording 2024-10-05 at 22 52 50](https://github.com/user-attachments/assets/bf9db2cc-0420-48dd-811b-e8a62f330179)

Running expo with the cache cleared (running without clearing the cache doesn't trigger the error in the very first execution after setting components.json)
![Recording 2024-10-05 at 22 58 39](https://github.com/user-attachments/assets/81664394-8b71-4de2-b8fd-93c02e77e2d5)

The video cuts were only due to the time it took to render it all

**Comment by mrzachnugent** [comment]
Hey @pedromchd, without a minimal reproduction repo, I cannot help unfortunately. I cannot reproduce the error on my side without it.

**Comment by pedromchd** [comment]
Hi! OMG, I'm so sorry, I got a busy day yesterday and somehow understood reprodution steps.... Here it is a reproduction repo: [Minimal RNR Test](https://github.com/pedromchd/mininal-rnr-test)

**Comment by mrzachnugent** [comment]
Hey @pedromchd, looks like the issue comes from `@/app/index.tsx`, here's what I changed to get it to work:

```diff
- import { ControlledInput, SafeAreaView } from '@/components';
+ import { ControlledInput } from '@/components/ControlledInput';
+ import { SafeAreaView } from '@/components/SafeAreaView';
```

Not sure why exactly, it might be circular dependencies or something which is out of scope of rnr.

**Comment by pedromchd** [comment]
I had a clue so I did some edits and ran a test again, and I presume that its a conflict between `components.json` and `components/index.ts`, you see, just as html, a index file can be ommited and have just the parent's folder name: `components`, so when Expo tries to render, there must be some confusion and it actually loads components.json instead of the folder. I got an error too when I created a file named "hooks.json", for example. During NativeWind install, a similliar warning says that you shouldn't name `nativewind.d.ts` with the same name as any other folder in the project, so its a mather of conflict in the naming, I solved this just removing the index.ts file from components folder.
